generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -2033,15 +2033,21 @@ private void generateClassEquals(String className, StructSpec struct,
             elementKeysAreEqual ? "elementKeysAreEqual" : "equals");
         buffer.incrementIndent();
         buffer.printf("if (!(obj instanceof %s)) return false;%n", className);
+        buffer.printf("%s other = (%s) obj;%n", className, className);
         if (!struct.fields().isEmpty()) {
-            buffer.printf("%s other = (%s) obj;%n", className, className);
             for (FieldSpec field : struct.fields()) {
                 if (!elementKeysAreEqual || field.mapKey()) {
                     generateFieldEquals(field);
                 }
             }
         }
-        buffer.printf("return true;%n");
+        if (elementKeysAreEqual) {
```

Comment:
What is the purpose of this?

Reply:
This function in `MessageDataGenerator` is used to generate `equals`, but it is also used to generate the `elementKeysAreEqual` function.  The latter function does not consider unknown tagged fields, since they can't be part of the key.

Reply:
Thanks for the clarification, makes sense.